sound/soc/sof/intel/hda-dsp.c

```diff
@@ -622,8 +622,7 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 	hda_dsp_ipc_int_disable(sdev);
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
-	if (runtime_suspend)
-		hda_codec_jack_wake_enable(sdev, true);
+	hda_codec_jack_wake_enable(sdev, runtime_suspend);
```

Comment:
For S3 case, is there a behavior the system will be woken up when hotplug is happening ?

Reply:
no, but we want to check the status of the jack when we resume.

Reply:
@RanderWang So this new version now aligns with prior state and the snd-hda-intel behavior, WAKEEN is not set when we go to S3. If we go to S3, runtime_suspend is false and hda_codec_jack_wake_enable() will clear WAKEEN.
If we go to runtime-suspend, runtime_suspend is true and  hda_codec_jack_wake_enable() will set WAKEEN bits.

So in short, no, we should not wake up the system for hotplug.